pkg/logutil/logutil_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package logutil
+
+import (
+	"bytes"
+	"os"
+	"testing"
+)
+
+func TestNewFileWriter(t *testing.T) {
+	for _, tt := range []struct {
+		name          string
+		dirPath       string
+		filename      string
+		maxSize       int
+		startContent  []byte
+		appendContent []byte
+		wantContent   []byte
+	}{
+		{
+			name:          "append to file",
+			dirPath:       "",
+			filename:      "file.log",
+			maxSize:       1024,
+			startContent:  []byte("foo"),
+			appendContent: []byte("bar"),
+			wantContent:   []byte("foobar"),
+		},
+		{
+			name:          "append to file too large",
+			dirPath:       "",
+			filename:      "file.log",
+			maxSize:       -1,
+			startContent:  []byte("foo"),
+			appendContent: []byte("bar"),
+			wantContent:   []byte("foo"),
+		},
+		{
+			name:          "append overflow",
+			dirPath:       "",
+			filename:      "file.log",
+			maxSize:       5,
+			startContent:  []byte("foo"),
+			appendContent: []byte("bar"),
+			wantContent:   []byte("fooba"),
+		},
+		{
+			name:          "dir missing",
+			dirPath:       "dir",
+			filename:      "file.log",
+			maxSize:       1024,
+			startContent:  []byte(""),
+			appendContent: []byte("bar"),
+			wantContent:   []byte("bar"),
+		},
```

Comment:
add test cases for covering line 23, and 28

Reply:
Not sure how to test line 28.. somehow make the stat fail?

Reply:
Worth checking implementation of Stat, and see which case will it return an error, and mock accordingly.